### PR TITLE
*: bypass redundant checks for binary point select

### PR DESCRIPTION
**Pull request by jackysp**
Signed-off-by: Shuaipeng Yu <jackysp@gmail.com>

### What problem does this PR solve? 
Point select does not need to start a transaction if it is an auto-committed statement.

### What is changed and how it works?
Bypass most redundant checks when it is an auto-committed point select statement in binary protocol.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

### CODE REVIEW

**Comment by jackysp** [comment]
Point select test:
master: 136k
this PR: 143k

File: executor/adapter.go [hunk 1]
```diff
@@ -734,15 +734,6 @@ func IsPointGetWithPKOrUniqueKeyByAutoCommit(ctx sessionctx.Context, p plannerco
 		return false, nil
 	}
 
-	// check txn
-	txn, err := ctx.Txn(false)
```
Comment (coocood) [inline executor/adapter.go]: Make we check also check that the session is not in a transaction by `SessionVars.InTxn`

File: executor/prepared.go [hunk 2]
```diff
@@ -268,20 +258,26 @@ func (e *DeallocateExec) Next(ctx context.Context, req *chunk.Chunk) error {
...
 	}
 	execStmt.BinaryArgs = args
 	is := GetInfoSchema(sctx)
 	execPlan, err := planner.Optimize(ctx, sctx, execStmt, is)
 	if err != nil {
-		return nil, err
+		return nil, false, err
+	}
+
+	useMaxTS, err := IsPointGetWithPKOrUniqueKeyByAutoCommit(sctx, execPlan.(*plannercore.Execute).Plan)
```
Comment (coocood) [inline executor/prepared.go]: We better rename the function to `CanUseMaxTS`.

File: executor/adapter.go (hunk 1 above)
Comment (jackysp) [reply executor/adapter.go]: good catch

**Comment by coocood** [comment]
LGTM

File: session/session.go [hunk 3]
```diff
@@ -1228,11 +1229,20 @@ func (s *session) PrepareStmt(sql string) (stmtID uint32, paramCount int, fields
...
 	if err != nil {
 		return nil, err
 	}
 	logQuery(st.OriginText(), s.sessionVars)
+	if useMaxTS {
+		err = s.InitTxnWithStartTS(math.MaxUint64)
+		if err != nil {
+			return nil, err
+		}
+		rs, err := st.Exec(ctx)
```
Comment (lysu) [inline session/session.go]: directly call this will miss some info like `session.runStmt` tracing?

**Pull request closed without merging**